makeflow/src/makeflow_module_archive.c

```diff
@@ -0,0 +1,503 @@
+			char * file_id = generate_file_archive_id(a, f);
+			sha1_update(&context, "I", 1);
+			sha1_update(&context, file_id, strlen(file_id));
+			sha1_update(&context, "\0", 1);
+			free(file_id);
+		}
+		list_cursor_destroy(cur);
+
+		/* Sort outputs for consistent hashing */
+		list_sort(t->output_files, batch_file_outer_compare);
+
+		/* add checksum of the node's output file names together */
+		cur = list_cursor_create(t->output_files);
+		for(list_seek(cur, 0); list_get(cur, (void**)&f); list_next(cur)) {
+			printf("output file : %s\n", f->outer_name);
+			sha1_update(&context, "O", 1);
+			sha1_update(&context, f->outer_name, strlen(f->outer_name));
+			sha1_update(&context, "\0", 1);
+		}
+		list_cursor_destroy(cur);
+
+		sha1_final(digest, &context);
+		itable_insert(a->task_ids, t->taskid, xxstrdup(sha1_string(digest)));
+	}
+	return xxstrdup(itable_lookup(a->task_ids, t->taskid));
+}
+
+/* writes the run_info files that is stored within each archived node */
+static int makeflow_archive_task_info(struct archive_instance *a, struct dag_node *n, struct batch_task *t, char *archive_path) {
+
+	struct batch_file *f;
+	char *task_info = string_format("%s/task_info", archive_path);
+
+	FILE *fp = fopen(task_info, "w");
+	if (fp == NULL) {
+		free(task_info);
+		debug(D_ERROR|D_MAKEFLOW_HOOK, "could not create task_info for node %d archive", n->nodeid);
+		return 1;
+	} else {
+		fprintf(fp, "COMMAND : %s\n", t->command);
+		fprintf(fp, "SRC_COMMAND : %s\n", n->command);
+		fprintf(fp, "SRC_MAKEFLOW : %s\n", a->source_makeflow);
+		fprintf(fp, "INPUT_FILES : \n");
+		struct list_cursor *cur = list_cursor_create(t->input_files);
+		for(list_seek(cur, 0); list_get(cur, (void**)&f); list_next(cur)) {
+			char *id = generate_file_archive_id(a, f);
+			fprintf(fp, "\t%s : %s\n", f->outer_name, id);
+			free(id);
+		}
+		list_cursor_destroy(cur);
+
+		fprintf(fp, "OUTPUT_FILES : \n");
+		cur = list_cursor_create(t->output_files);
+		for(list_seek(cur, 0); list_get(cur, (void**)&f); list_next(cur)) {
+			char *id = generate_file_archive_id(a, f);
+			fprintf(fp, "\t%s : %s\n", f->outer_name, id);
+			free(id);
+		}
+		list_cursor_destroy(cur);
+	}
+	fclose(fp);
+	free(task_info);
+
+	task_info = string_format("%s/run_info", archive_path);
+
+	fp = fopen(task_info, "w");
+	if (fp == NULL) {
+		free(task_info);
+		debug(D_ERROR|D_MAKEFLOW_HOOK, "could not create run_info for node %d archive", n->nodeid);
+		return 1;
+	} else {
+		fprintf(fp, "SUBMITTED : %" PRIu64 "\n", t->info->submitted);
+		fprintf(fp, "STARTED : %" PRIu64 "\n", t->info->started);
+		fprintf(fp, "FINISHED : %" PRIu64 "\n", t->info->finished);
+		fprintf(fp, "EXIT_NORMAL : %d\n", t->info->exited_normally);
+		fprintf(fp, "EXIT_CODE : %d\n", t->info->exit_code);
+		fprintf(fp, "EXIT_SIGNAL : %d\n", t->info->exit_signal);
+	}
+	fclose(fp);
+	free(task_info);
+
+	return 0;
+}
+
+/* writes the file symlink that links to the archived job that created it */
```

Comment:
I would update this comment to be more reflective of the function, or change the name of the function, or do both! The function does checksum, yes, but it also does a more, but the checksum isn't really talked about in the comment.